### PR TITLE
Form validation in register form

### PR DESCRIPTION
**Pull request by pkakelas**
Also please notice the regexp in the username validation. I suggest to use this length validation from now on

### CODE REVIEW

File: static/script/form_validation.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+$( document ).ready( function() { 
+    $( '#register-form' ).submit(function() {
+        var $username = $( '#username' ).val();
```
Comment (dionyziz) [inline static/script/form_validation.js]: Please don't prefix regular variables with `$`. Use this notation only for variables that store jQuery objects.

File: static/script/form_validation.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+$( document ).ready( function() { 
+    $( '#register-form' ).submit(function() {
```
Comment (dionyziz) [inline static/script/form_validation.js]: nit: space after `(`

File: static/script/form_validation.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        if ( $password == '' ) {
+            createError( '#register-form', 'Please type a password.' );
+            return false;
+        }
+        if ( $password.length < 7 ) {
+            createError( '#register-form', 'Password should be at least 7 characters long' );
+            return false;
+        }
+        if ( $passwordRepeat != $password ) {
+            createError( '#register-form', 'Please repeat correctly your password' );
```
Comment (dionyziz) [inline static/script/form_validation.js]: "Passwords do not match"

File: static/script/form_validation.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            return false;
+        }
+        if ( $email == '' ) {
+            createError( '#register-form', 'Please type an email.' );
+            return false;
+        }
+    }); 
+
+    function createError( $form, $description ) {
+        if ( $('.alert').length != 0 ) {
```
Comment (dionyziz) [inline static/script/form_validation.js]: nit: spaces within `()`

File: static/script/form_validation.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            $( '.alert' ).remove();
```
Comment (dionyziz) [inline static/script/form_validation.js]: No need for `length` check; `remove` only removes if they exist.

**Comment by bugos** [comment]
Why don't we use HTML5 form validation?

**Comment by dionyziz** [comment]
That's a good call. Maybe we should consider it. @pkakelas what are your thoughts on @bugos' suggestion?

Incidentally, here's a tutorial:

http://www.sitepoint.com/html5-form-validation/

**Comment by pkakelas** [comment]
I've already made a research about html5 form validation as @gtklocker had
also made the suggestion some weeks ago. I used this kind of validation
mainly because it was a good motive to practice my JavaScript skills. If
you want, I can change the validation, but I prefer the way it is as we can
have all the validation code in one place.
Στις 29 Απρ 2014 12:48 π.μ., ο χρήστης "Dionysis Zindros" <
notifications@github.com> έγραψε:

**Comment by dionyziz** [comment]
Agreed, let's stay with JS validation.

File: static/script/form_validation.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        var $username = $( '#username' ).val();  
+        var $password = $( '#password' ).val();  
+        var $passwordRepeat = $( '#password_repeat' ).val();  
+        var $email = $( '#email' ).val();  
+        if ( $username == '' ) {
+            createError( '#register-form', 'Please type a username.' );
+            return false;
+        }
+        if ( !$username.match( /^[a-zA-Z0-9_.]{3,16}$/ ) ) {
+            createError( '#register-form', 'Usernames can only have numbers, letters, "." and "_"' );
```
Comment (dionyziz) [inline static/script/form_validation.js:12]: This regexp works well for validation, but we are unable to provide a meaningful error message. For example, if the user enters a 2-character username, the error message is quite confusing.

File: static/script/form_validation.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            return false;
+        }
+        $.get(
+            'user/view',
+            {
+                username: $username 
+            },
+            function( responseText ) {
+                    createError( '#register-form', 'Username already exists' );
+                    return false;
```
Comment (dionyziz) [inline static/script/form_validation.js]: nit: use 4 spaces for indentation

**Comment by dionyziz** [comment]
Screenshots please.

File: static/script/form_validation.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    $( '#register-form' ).submit( function() {
+        var username = $( '#username' ).val();  
+        var password = $( '#password' ).val();  
+        var passwordRepeat = $( '#password_repeat' ).val();  
+        var email = $( '#email' ).val();  
+        if ( username == '' ) {
+            createError( '#register-form', 'Please type a username' );
+            return false;
+        }
+        if ( !username.match( /^\s*[a-zA-Z0-9,\s._]+\s*$/ ) ) {
```
Comment (dionyziz) [inline static/script/form_validation.js]: Why do you allow whitespace around usernames?

File: static/script/form_validation.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+            return false;
+        }
+        $.get(
+            'user/view',
+            {
+                username: username 
+            },
+            function( responseText ) {
+                createError( '#register-form', 'Username already exists' );
+                return false;
```
Comment (dionyziz) [inline static/script/form_validation.js]: Why is return needed here?

File: static/script/form_validation.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+            },
+            "html"
+        );
+        if ( password == '' ) {
+            createError( '#register-form', 'Please type a password' );
+            return false;
+        }
+        if ( password.length < 7 ) {
+            createError( '#register-form', 'Passwords do not match' );
```
Comment (dionyziz) [inline static/script/form_validation.js]: Error message doesn't seem to correspond to if condition.

File: static/script/form_validation.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+            return false;
+        }
+        if ( passwordRepeat != password ) {
+            createError( '#register-form', 'Please repeat correctly your password' );
```
Comment (dionyziz) [inline static/script/form_validation.js]: "Passwords do not match"

**Comment by dionyziz** [comment]
Please provide a screenshot.

**Comment by pkakelas** [comment]
Nothing has changed. It's just like the back end validation.
Στις 30 Απρ 2014 4:21 μ.μ., ο χρήστης "Dionysis Zindros" <
notifications@github.com> έγραψε:

**Pull request merged**